snapcraft/cli/__init__.py

```diff
@@ -89,6 +89,7 @@ def get_command(self, ctx, cmd_name):
 @click.option('--debug', '-d', is_flag=True)
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
+    ctx.obj['debug'] = debug
     if debug:
```

Comment:
What's it needed for if we're getting debug from kwargs? I don't see it being used for anything else. It contained the ProjectOptions before and even in master I don't see anything using obj. Is it indirectly being used some other way?